api/v1alpha1/storageclassclaim_types.go

```diff
@@ -29,9 +29,18 @@ type StorageClassClaimSpec struct {
 	EncryptionMethod string `json:"encryptionMethod,omitempty"`
 }
 
+type storageClassClaimState string
+
+const (
+	StorageClassClaimCreating    storageClassClaimState = "Creating"
+	StorageClassClaimConfiguring storageClassClaimState = "Configuring"
+	StorageClassClaimReady       storageClassClaimState = "Ready"
```

Comment:
We dont have failed state?

Reply:
A filed state is not an error state, each one of the state can return an error.
A failed state is for an error in the desired state that cannot be recovered without a change to the desired state.
I don't think we have it for this controller